.github/workflows/ci.yml

```diff
@@ -1,7 +1,13 @@
 name: CI
 on: [push, pull_request]
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
```

Comment:
can you just explain me what this does? already all jobs run in parallel so what does this solve?

Reply:
it means if same pr committed multiple times, will just cancel the ongoing test, if have any,just run the latest one instead